clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -162,14 +168,23 @@ func (s *Sequencer) batchingLoop() {
 			}
 			if len(localTxs) > 0 {
 				sortedTxs := types.NewTransactionsByPriceAndNonce(signer, localTxs, batcher.header.BaseFee)
-				batchTxs = s.addTxsToBatchAndCommit(batcher, sortedTxs, batchTxs, signer)
+				batchTxs, err = s.addTxsToBatchAndCommit(batcher, sortedTxs, batchTxs, signer)
+				if err != nil {
+					log.Crit("Failed to add transaction to batch and commit", "err", err)
```

Comment:
imo the callee should provide this context (e.g. you already have `Failed to commit transactions` wrapping one error, so you just need to wrap the other error similarly when `modifyTxnsInBatch` fails)

then here we can add higher level context here:
`log.Crit("batchingLoop failed to process local txs: %w", "err", err)`

same for below

Reply:
yes, what i mean is the line here can just be something like:
`log.Crit("batchingLoop failed to process local txs", "err", err)`. you don't need to provide the context you've currently provided because it's included in `err`

similarly, in L179 you can do: `log.Crit("batchingLoop failed to process remote txs", "err", err)`
in L200: `log.Crit("Failed to process txsCh event ", "err", err)`

i.e. i think we need to be consistent around the level of the stack for which context is being provided (caller-level vs callee-level)

Reply:
I added logs in format of [File:Function] so just by error we can pin point source of error.